### PR TITLE
Gradle improvements + updates

### PR DESCRIPTION
**Pull request by hzsweers**

### CODE REVIEW

**Comment by hzsweers** [comment]
I'm not sure what the failure is from, maybe a permissions error for spoon?

**Comment by JayNewstrom** [comment]
See my PR #251 I think the failure is an out of memory.

File: build.gradle [hunk 1]
```diff
@@ -132,18 +132,28 @@ configurations {
...
   }
 }
 
 dependencies {
-  compile 'com.android.support:support-v4:23.3.0'
-  compile 'com.android.support:support-annotations:23.3.0'
-  compile 'com.android.support:appcompat-v7:23.3.0'
-  compile 'com.android.support:recyclerview-v7:23.3.0'
-  compile 'com.android.support:design:23.3.0'
+  supportCompile "com.android.support:support-v4:$supportLibraryVersion"
```
Comment (jaredsburrows) [inline build.gradle]: Nice

**Comment by JakeWharton** [comment]
Most of this is outdated now. Sorry for sitting on it.

**Pull request closed without merging**